<?xml version="1.0" encoding="utf-8"?>
<launch>

  <!-- Arguments -->
  <arg name="use_sim_time" default="false"/>

  <arg name="cmd_vel_topic" default="/omni_base_controller/cmd_vel"/>
  <arg name="laser_scan_topic" default="/scan"/>
  <arg name="laser_scan_frame" default="base_range_sensor_link"/>
  <arg name="legs_pose_topic" default="/hri/leg_finder/leg_pose"/>
  <arg name="head_topic" default="/hardware/head/goal_pose"/>

  <arg name="move_head" default="false"/>
  <arg name="use_pot_fields" default="true"/>

  <group>
      <push-ros-namespace namespace="hri"/>
      <!-- Leg Finder -->
      <node name="leg_finder" 
            pkg="human_follower" 
            exec="leg_finder_node" 
            output="screen"
            respawn="true">
            <param name="scan_downsampling" value="1"/>
            <param name="show_hypothesis" value="false"/>

            <param name="laser_scan_frame" value="$(var laser_scan_frame)"/>
            <param name="laser_scan_topic" value="$(var laser_scan_topic)"/>
      </node>

      <!-- Human Follower -->
      <node name="human_follower" 
            pkg="human_follower" 
            exec="human_follower_node" 
            output="screen"
            respawn="true">
            <param name="control_alpha" value="0.8"/>
            <param name="control_beta" value="0.3"/>
            <param name="max_linear" value="0.7"/>
            <param name="max_angular" value="0.7"/>
            <param name="dist_to_human" value="0.9"/>

            <param name="move_backwards" value="false"/>
            <param name="move_head" value="$(var move_head)"/>
            <param name="pot_fields" value="$(var use_pot_fields)"/>

            <param name="legs_pose_topic" value="$(var legs_pose_topic)"/>
            <param name="cmd_vel_topic" value="$(var cmd_vel_topic)"/>
            <param name="head_topic" value="$(var head_topic)"/>
      </node>
  </group>

</launch>
